kv: all leases start as expiration based

andrewbaptist opened the pull request:
Starting a lease as an epoch lease is risky. The problem is the lease may be marked as transferred to the new node, but the node can take time to realize it is the leaseholder. THis was partially addressed as part of #85629, however there were still cases where this can occur. Specifically in the cases of non-cooperative lease acquisition. While we don't have an example of this causing a problem because the range is likely idle at this point, it is certainly possible to make a contrived case where this happens.

Epic: none

Release note: None

Commit:
kv: all leases start as expiration based

Starting a lease as an epoch lease is risky. The problem is the lease
may be marked as transferred to the new node, but the node can take time
to realize it is the leaseholder. THis was partially addressed as part
of #85629, however there were still cases where this can occur.
Specifically in the cases of non-cooperative lease acquisition. While
we don't have an example of this causing a problem because the range is
likely idle at this point, it is certainly possible to make a contrived
case where this happens.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/101765)
<!-- Reviewable:end -->

Commit:
kv: Improve metrics test

The metrics are a little confusing since they are measured as a delta
since the last time the call is made in some cases, but in other cases
they seem to be cumulative. Need to figure this out more still....
